sdk/storage/inc/adls/protocol/adls_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+  {
+    if (pathGetPropertiesAction == "getAccessControl")
+    {
+      return PathGetPropertiesAction::GetAccessControl;
+    }
+    if (pathGetPropertiesAction == "getStatus")
+    {
+      return PathGetPropertiesAction::GetStatus;
+    }
+    throw std::runtime_error("Cannot convert " + pathGetPropertiesAction + " to PathGetPropertiesAction");
+  };  // enum class PathGetPropertiesAction
+
+  struct ServiceListFileSystemsOptions
+  {
+    std::string Prefix = std::string();  // Filters results to filesystems within the specified prefix.
+    std::string Continuation = std::string();  // Optional.  When deleting a directory, the number of paths that are deleted with each invocation is limited.  If the number of paths to be deleted exceeds this limit, a continuation token is returned in this response header.  When a continuation token is returned in the response, it must be specified in a subsequent invocation of the delete operation to continue deleting the directory.
+    uint32_t MaxResults = uint32_t();  // An optional value that specifies the maximum number of items to return. If omitted or greater than 5,000, the response will include up to 5,000 items.
+    std::string ClientRequestId = std::string();  // Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the analytics logs when storage analytics logging is enabled.
+    uint32_t Timeout = uint32_t();  // The timeout parameter is expressed in seconds. For more information, see <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations">Setting Timeouts for Blob Service Operations.</a>
+    std::string ApiVersionParameter = k_DefaultServiceApiVersion;  // Specifies the version of the operation to use for this request.
+  };
+
+  static Azure::Core::Http::Request ServiceListFileSystemsCreateRequest(std::string url, const ServiceListFileSystemsOptions& serviceListFileSystemsOptions)
+  {
+    Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Get, url);
+    request.AddQueryParameter(k_QueryResource, "account");
+    if (!serviceListFileSystemsOptions.Prefix.empty())
+    {
+      request.AddQueryParameter(k_QueryPrefix, serviceListFileSystemsOptions.Prefix);
+    }
+    if (!serviceListFileSystemsOptions.Continuation.empty())
+    {
+      request.AddQueryParameter(k_QueryContinuation, serviceListFileSystemsOptions.Continuation);
+    }
+    if (!std::to_string(serviceListFileSystemsOptions.MaxResults).empty())
+    {
+      request.AddQueryParameter(k_QueryMaxResults, std::to_string(serviceListFileSystemsOptions.MaxResults));
+    }
+    if (!serviceListFileSystemsOptions.ClientRequestId.empty())
+    {
+      request.AddHeader(k_HeaderClientRequestId, serviceListFileSystemsOptions.ClientRequestId);
+    }
+    if (!std::to_string(serviceListFileSystemsOptions.Timeout).empty())
```

Comment:
I think it's never empty.

Reply:
ditto for other integer variables

Reply:
This is true, I'll mark this as something to change for Optional<T>.